lib/logstash/filters/dns.rb

```diff
@@ -122,6 +122,12 @@ def resolve(event)
     @resolve.each do |field|
       is_array = false
       raw = event.get(field)
+
+      if raw.nil?
+        @logger.warn("DNS filter could not resolve missing field", :field => field)
```

Comment:
my main concern with this strategy is that if a user provides a list of 3 fields and one is nil, then the filter will still be successful, but pass an event further with a nil without any tags.

I think it's worth going to the extra effort of tagging and breaking out of the loop in case of a nil reference